color/src/css.rs

```diff
@@ -124,6 +124,20 @@ impl CssColor {
         }
     }
 
+    /// Clip the color's components to the range allowed by the colorspace.
+    ///
+    /// See [`Colorspace::clip`] for more details.
+    #[must_use]
+    pub fn clip(self) -> Self {
+        let (opaque, alpha) = split_alpha(self.components);
+        let components = self.cs.clip(opaque);
+        Self {
+            cs: self.cs,
+            missing: self.missing,
+            components: add_alpha(components, alpha),
+        }
+    }
+
```

Comment:
Alpha is unchanged here, maybe it should be clamped to [0, 1].

Reply:
I think so, yes.

Also a really subtle point I'll raise here, that doesn't need change in the code. There is a not-strictly-enforced invariant that missing channels have 0.0 components. Since clipping will always preserve 0, there's no fixup needed here.

Reply:
The alpha is now also clamped.

I did not copy the set-missing-to-0 from the `scale_chroma` function above for that reason; it may be worth documenting the invariant itself.